packages/integration_test/lib/_callback_io.dart

```diff
@@ -54,9 +59,43 @@ class IOCallbackManager implements CallbackManager {
     // comes up in the future. For example: `WebCallbackManager.cleanup`.
   }
 
+  // Whether the Flutter surface uses an Image.
+  bool _usesFlutterImage = false;
+
   @override
-  Future<void> takeScreenshot(String screenshot) {
-    throw UnimplementedError(
-        'Screenshots are not implemented on this platform');
+  Future<void> convertFlutterSurfaceToImage() async {
+    await integrationTestChannel.invokeMethod<void>(
+      'convertFlutterSurfaceToImage',
+      null,
+    );
+    _usesFlutterImage = true;
+  }
+
+  @override
+  Future<Map<String, dynamic>> takeScreenshot(String screenshot) async {
+    if (!_usesFlutterImage) {
+      throw 'Please call convertFlutterSurfaceToImage() before taking a screenshot';
+    }
+    integrationTestChannel.setMethodCallHandler(_onMethodChannelCall);
+    final List<int>? rawBytes = await integrationTestChannel.invokeMethod<List<int>>(
+      'captureScreenshot',
+      null,
+    );
+    if (rawBytes == null) {
+      throw 'Expected a list of bytes, but instead captureScreenshot returned null';
+    }
+    return <String, dynamic>{
+      'screenshotName': screenshot,
+      'bytes': rawBytes,
+    };
+  }
+
+  Future<dynamic> _onMethodChannelCall(MethodCall call) {
+    switch (call.method) {
+      case 'scheduleFrame':
+        window.scheduleFrame();
+        break;
+    }
+    return Future<dynamic>.value(null);
```

Comment:
nit: if the method is marked as `async` this won't be needed, which is a little cleaner

Reply:
done